plugins/catalog-backend/src/database/DefaultProcessingDatabase.ts

```diff
@@ -108,9 +108,20 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
     });
 
     // Delete old relations
-    await tx<DbRelationsRow>('relations')
-      .where({ originating_entity_id: id })
-      .delete();
+    let previousRelationRows: DbRelationsRow[];
+    if (tx.client.config.client.includes('sqlite3')) {
+      previousRelationRows = await tx<DbRelationsRow>('relations')
+        .select('*')
+        .where({ originating_entity_id: id });
+      await tx<DbRelationsRow>('relations')
+        .where({ originating_entity_id: id })
+        .delete();
+    } else {
+      previousRelationRows = await tx<DbRelationsRow>('relations')
```

Comment:
There has to be a race here right? ^_> Thinking one entity having its relations updated, with a related one being stitched at the same time. The other one might read the relations first before deletion, but still winning the stitching afterwards.